odbc/src/api.rs

```diff
@@ -800,13 +801,41 @@ pub extern "C" fn SQLGetEnvAttr(
 
 #[no_mangle]
 pub extern "C" fn SQLGetEnvAttrW(
-    _environment_handle: HEnv,
-    _attribute: EnvironmentAttribute,
-    _value_ptr: Pointer,
+    environment_handle: HEnv,
+    attribute: EnvironmentAttribute,
+    value_ptr: Pointer,
     _buffer_length: Integer,
-    _string_length: *mut Integer,
-) -> SqlReturn {
-    unimplemented!()
+    string_length: *mut Integer,
+) -> SqlReturn {
+    let env_handle = MongoHandleRef::from(environment_handle);
+    env_handle.clear_diagnostics();
+    match env_handle.as_env() {
+        None => SqlReturn::INVALID_HANDLE,
+        Some(env) => {
+            let env_contents = env.read().unwrap();
+            unsafe {
+                *string_length = match value_ptr.is_null() {
+                    true => 0,
+                    false => size_of::<Integer>() as Integer,
+                }
+            }
+            match attribute {
```

Comment:
Are you sure that this will not blow up if value_ptr is null? Is there some kind of Rust magic?

Also, value_ptr is a pointer to a buffer the user/app provides to store the output data. If a null pointer is passed, how can we can change it. If we were to provide the pointer to the output buffer it would work, but it is not the case.

Reply:
Ah, looks like I'm missing a null pointer check! I'll add that in